test/unit/db/repos/blocks.js

```diff
@@ -0,0 +1,885 @@
+					limit: 3,
+				});
+
+				expect(data).to.be.not.empty;
+				expect(data).to.be.an('array');
+				// There are 5 total blocks in the seed
+				return expect(data).to.be.lengthOf(2);
+			});
+		});
+
+		describe('loadBlocksOffset()', () => {
+			it('should use the correct SQL file two parameters', function*() {
+				sinonSandbox.spy(db, 'query');
+
+				yield db.blocks.loadBlocksOffset(10, 20);
+
+				expect(db.query.firstCall.args[0]).to.eql(sql.loadBlocksOffset);
+				expect(db.query.firstCall.args[1]).to.eql([10, 20]);
+				return expect(db.query).to.be.calledOnce;
+			});
+
+			it('should not be rejected if param "offset" is missing', () => {
+				return expect(
+					db.blocks.loadBlocksOffset(null, 10)
+				).to.be.eventually.eql([]);
+			});
+
+			it('should not be rejected if param "limit" is missing', () => {
+				return expect(db.blocks.loadBlocksOffset(0, null)).to.be.eventually.eql(
+					[]
+				);
+			});
+
+			it('should return data with valid format', function*() {
+				const data = yield db.blocks.loadBlocksOffset(0, 5);
+
+				expect(data).to.be.not.empty;
+				expect(data).to.be.an('array');
+				expect(data[0]).to.have.all.keys(blockListFields);
+
+				return data.forEach(block => {
+					expect(block.b_height).to.be.above(0);
+					expect(block.b_height).to.be.below(5);
+				});
+			});
+		});
+
+		describe('loadLastBlock()', () => {
+			it('should use the correct SQL file with no parameters', function*() {
+				sinonSandbox.spy(db, 'query');
+
+				yield db.blocks.loadLastBlock();
+
+				expect(db.query.firstCall.args[0]).to.eql(sql.loadLastBlock);
+				expect(db.query.firstCall.args[1]).to.eql(undefined);
+				return expect(db.query).to.be.calledOnce;
+			});
+
+			it('should return last block with valid data format', function*() {
+				const data = yield db.blocks.loadLastBlock();
+
+				expect(data).to.be.not.empty;
+				expect(data).to.be.an('array');
+				expect(data).to.have.lengthOf(1);
+				expect(data[0]).to.have.all.keys(blockListFields);
+
+				// As we only see 5 blocks
+				return expect(data[0].b_height).to.be.eql(5);
+			});
+		});
+
+		describe('loadLastNBlockIds()', () => {
+			it('should use the correct SQL file with one parameter', function*() {
+				sinonSandbox.spy(db, 'query');
+
+				yield db.blocks.loadLastNBlockIds(1);
+
+				expect(db.query.firstCall.args[0]).to.eql(sql.loadLastNBlockIds);
+				expect(db.query.firstCall.args[1]).to.eql([1]);
+				return expect(db.query).to.be.calledOnce;
+			});
+
+			it('should return last block ids', function*() {
+				const genesisBlock = blocksFixtures.GenesisBlock();
+				const data = yield db.blocks.loadLastNBlockIds(5);
+
+				expect(data).to.be.not.empty;
+				expect(data).to.be.an('array');
+				expect(data).to.have.lengthOf(5); // As look for 5 blocks
+				expect(data[0]).to.have.all.keys('id');
+
+				// Due to descending order on height genesis block will appear in last
+				return expect(data[data.length - 1].id).to.be.eql(genesisBlock.id);
```

Comment:
can we test all data set here?

Reply:
We don't keep the track of fixtures we load during seed. We need to refactor that seeding logic to test all blocks.